File: 1200.minimum-absolute-difference/solution.go
func minimumAbsDifference(arr []int) [][]int {
    slices.Sort(arr)

    m := math.MaxInt
    for i := 1; i < len(arr); i++ {
        m = min(m, arr[i] - arr[i - 1])
    }

    var res [][]int
    for i := 1; i < len(arr); i++ {
        if arr[i] - arr[i - 1] == m {
            res = append(res, []int{arr[i - 1], arr[i]})
        }
    }

    return res
}

File: 1200.minimum-absolute-difference/solution.kt
class Solution {
    fun minimumAbsDifference(arr: IntArray): List<List<Int>> {
        val windowed = arr.sorted().windowed(2)
        val min = windowed.map { it[1] - it[0] }.min()
        return windowed.filter { it[1] - it[0] == min }.map { listOf(it[0], it[1]) }
    }
}